### DISCUSSION TITLE
`m` doesn't select to the matching enclosing under the cursor.

### DISCUSSION THREAD

**Discussion by leira** [71/dance#227] [General]
Key `m` doesn't behaves the same as in Kakoune. This is an example:

```
{ a { b } c }
----------------
^ the cursor is here
``` 

Currently, the cursor is on the first `{`. In Kakoune, if `m` is pressed, it will select from the first `{` to the last `}` after `c`, and the cursor stops at the last `}`. When `m` is pressed again, the selection won't change, the cursor jumps back to the first `{`. Keep pressing `m`s, the selection stays the same, the cursor ping-pongs between the 2 ends of the selection.

In Dance, if `m` is pressed, it selects from the second `{` after `a` to the first `}` after `b`, and the cursor stops at the first `}`. When `m` is pressed again, it selects the from the first `{` to the last `}`, and cursor jumps to the first `{`. The selection changes. And keep pressing `m`s, the selection keeps changing.

It seems, rather than starting from the character under the cursor to find a character to enclose with, Dance starts from the next character. Is this behavior difference intentional? Is there some technical limitations?

**Comment by leira** [comment]
I saw this comment in function enclosing(). It looks like it is trying to be consistent with Kakoune. And the details of the behavior matches my understanding of the Kakoune `m`.

https://github.com/71/dance/blob/9265b42b4c0170549adfd468574fca1598a36cd9/src/commands/seek.ts#L137-L141

**Reply by 71** [reply]
Indeed, there was a problem in the implementation. There are still some buggy edge cases, but 6603df95ea87bfa184ea74625e25fc0515e65077 fixes some of them.

**Comment by leira** [comment]
Thanks! Is there an issue opened to document these edge cases?

**Reply by 71** [reply]
Not yet, no. Feel free to open one (preferably with [test cases](https://github.com/71/dance/tree/master/test#command-tests)) if you hit any of them.
